### ISSUE TITLE
Rule does not detect destructuring assignment in loops

### ISSUE THREAD

**Issue by angryzor** [AsaAyers/coffeelint-undefined-variables#1]
Rule recognizes variables defined through normal destructuring assignment, but not those defined through destructuring assignment in loop statements.

Example code:

``` coffeescript
for id, {foo, bar} of @myBlort
  console.log "Doing something with #{id}"
  foo.doSomething(bar)

for {noo, nar} in @myBord
  console.log "Doing something with #{id}"
  noo.doSomething(nar)

{a, b} = @teehee
console.log a + b
```

Coffeelint output:

```
  ✗ errors.coffee
     ✗ #3: Undefined variable. foo.
     ✗ #3: Undefined variable. bar.
     ✗ #7: Undefined variable. noo.
     ✗ #7: Undefined variable. nar.

✗ Lint! » 4 errors and 0 warnings in 1 file
```

**Comment by angryzor** [comment]
If statement doesn't seem to have this problem, though i can't think of a good use case of destructuring assignment there:

``` coffeescript
if {coo, moo} = @boo
  coo.doSomething(moo)
```

```
  ✓ errors.coffee

✓ Ok! » 0 errors and 0 warnings in 1 file
```
